cpp/src/io/parquet/writer_impl.cu

```diff
@@ -1160,21 +1163,21 @@ void writer::impl::write(table_view const& table)
     init_page_fragments(fragments, col_desc, num_rows, max_page_fragment_size);
   }
 
-  size_t global_rowgroup_base = md.row_groups.size();
+  auto const global_rowgroup_base = static_cast<size_type>(md.row_groups.size());
 
   // Decide row group boundaries based on uncompressed data size
-  size_t rowgroup_size   = 0;
-  uint32_t num_rowgroups = 0;
-  for (uint32_t f = 0, global_r = global_rowgroup_base, rowgroup_start = 0; f < num_fragments;
+  size_t rowgroup_size    = 0;
+  size_type num_rowgroups = 0;
+  for (size_type f = 0, global_r = global_rowgroup_base, rowgroup_start = 0; f < num_fragments;
        f++) {
     size_t fragment_data_size = 0;
     // Replace with STL algorithm to transform and sum
     for (auto i = 0; i < num_columns; i++) {
       fragment_data_size += fragments[i][f].fragment_data_size;
     }
```

Comment:
It would be nice to implement the TODO in this comment with a `std::accumulate`. It would be cleanest to first add a `begin()` and `end()` to the [`hostdevice_2dvector` here](https://github.com/rapidsai/cudf/blob/42ffe781dc06e976adb3e3a70084328d95f0fb2b/cpp/src/io/utilities/hostdevice_vector.hpp#L149). I would follow the semantics of these methods for `hostdevice_vector`:
https://github.com/rapidsai/cudf/blob/42ffe781dc06e976adb3e3a70084328d95f0fb2b/cpp/src/io/utilities/hostdevice_vector.hpp#L89-L92

Reply:
Discussed offline - this would be easier if the indexing order was reversed for this 2D data structure. For now I'm fine with taking no action here.